src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs

```diff
@@ -265,6 +265,32 @@ protected override DbCommand CreateDbCommand()
             => CreateCommand();
 
         /// <summary>
+        /// Create custom collation.
+        /// </summary>
+        /// <param name="name">Name of the collation.</param>
+        /// <param name="comparison">Method that compares two strings.</param>
+        public virtual void CreateCollation(string name, Comparison<string> comparison)
```

Comment:
We should probably  validate `name` too if the native API gives its "misuse" error for `null` or empty values.